tests/cuda_custom_call/cuda_custom_call_test.py

```diff
@@ -0,0 +1,227 @@
+# This test is intentionally structured to stay close to what a standalone JAX
+# custom call integration might look like. JAX test harness is in a separate
+# section towards the end of this file. The test can be run standalone by typing
+# "make" in the directory containing this file.
+
+import ctypes
+
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+from jax.extend import ffi
+from jax.lib import xla_client
+from jax._src.lib.mlir import ir
+from jaxlib.hlo_helpers import custom_call
+from jax.interpreters import mlir
+
+# start test boilerplate
+from absl.testing import absltest
+from jax._src import config
+from jax._src import test_util as jtu
+
+from absl import flags
+# escape hatch for running this example standalone.
+# note, paths to files in current working directory must be prefixed with "./".
+flags.DEFINE_string(
+    "jax_cuda_custom_call_shared_library",
+    # location where Bazel puts the runtime data dependency libfoo.so
+    "tests/cuda_custom_call/libfoo.so",
+    "Path to the custom call shared library.")
+config.parse_flags_with_absl()
+FLAGS = flags.FLAGS
```

Comment:
Does this need to be a flag? Could we use `__file__` to find the appropriate path?

Reply:
A flag is more explicit and flexible, is there a reason to prefer `__file__` over a flag?

Reply:
Moving the example to `docs` highlighted that using `__file__` is indeed more straightforward here (though less explicit).